internal/api/handler/achievement_handler.go

```diff
@@ -138,3 +134,35 @@ func ListAchievementsHandler(ctx context.Context, achSvc service.Service) func(h
 		})
 	}
 }
+
+// DeleteAchievementHandler returns an HTTP handler that deletes particular achievement using profileSvc.
+func DeleteAchievementHandler(ctx context.Context, achSvc service.Service) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		profileID, achID, err := helpers.GetMultipleParams(r)
+		if err != nil {
+			middleware.ErrorResponse(w, http.StatusBadGateway, err)
+			zap.S().Error("error while getting the IDs from request")
+			return
+		}
+		// Decode into the struct
+		req := decodeDeleteAchievementRequest(profileID, achID)
+
+		// call the service
+		err = achSvc.DeleteAchievement(ctx, req)
+		if err != nil {
+			if err == errors.ErrNoData {
+				middleware.SuccessResponse(w, http.StatusOK, specs.MessageResponse{
```

Comment:
I think no need to return an error if record not found while deletion, 
just log the warning msg no record found.

@Payal13 what you think?

Reply:
discussed with payal they told to keep as it but change the message